### PR TITLE
DCES-77: Fixed authentication issue in Scheduler

### PR DESCRIPTION
**Pull request by tanovellino**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-XXX)

Fixed issue with Scheduler connecting to MAAT API. Changes are basically in classes `MaatApiWebClientFactory` and `DcesReportAuthenticationConfig`.

The rest of the changes are amending TODO comments (to reflect to the actual tickets they should be addressed).

So it seems long but it is not many changes

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/config/DcesReportAuthenticationConfig.java [hunk 1]
```diff
@@ -10,27 +11,35 @@
 @Configuration
 @EnableWebSecurity
 public class DcesReportAuthenticationConfig {
+    public static final String API_PATH_CONTRIBUTIONS = "/api/internal/v1/dces/report/contributions/**";
```
Comment (LAA-Dave-Edwards) [inline dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/config/DcesReportAuthenticationConfig.java:14]: Worth documenting these in the readme, or somewhere else in the mean-time. So it doesn't get lost.

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/config/DcesReportAuthenticationConfig.java [hunk 2]
```diff
@@ -10,27 +11,35 @@
...
+                    .requestMatchers(HttpMethod.GET, API_PATH_CONTRIBUTIONS).permitAll()
+                    .requestMatchers(HttpMethod.GET, API_PATH_FDC).permitAll()
+                    // TODO (DCES-77): if authentication is required leave authenticated(), otherwise maybe change to deny
+                    .anyRequest().authenticated()
             )
             .sessionManagement(
-                sess -> sess.sessionCreationPolicy(SessionCreationPolicy.STATELESS)
-// TODO (DCES-67): confirm what level of authentication security will be required to access this service
-//            )
+                    sess -> sess.sessionCreationPolicy(SessionCreationPolicy.STATELESS)
```
Comment (LAA-Dave-Edwards) [inline dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/config/DcesReportAuthenticationConfig.java:36]: Can we change this to "session", or "sessionManagementConfigurer" if we're going to have sess?

File: dces-report-service/src/main/resources/application-connectiontest.yaml [hunk 3]
```diff
@@ -52,13 +52,8 @@ spring:
...
               - ${MAAT_API_OAUTH_SCOPE}
-  # This is only needed to add support for executable beans (e.g. for serverless and AWS Lambada)
-  cloud:
-    function:
-      web:
-        path: /report
   scheduling:
     contributions:
-      cron: "0-30/16 */1 * * * ?" # "0 0 5 5 * ?" (5th of every month at 05:00am)
+      cron: "0-30/16 */5 * * * ?"
```
Comment (LAA-Dave-Edwards) [inline dces-report-service/src/main/resources/application-connectiontest.yaml:57]: Keeping a comment showing intended behaviour isn't a bad idea here.

**Review by vloruncg — APPROVED** [review]
I'm not clear on all the comments, I believe they indicate that they will resolved with additional iterations. so approving the work in the respective files for the proxy service and config.

File: dces-report-service/src/main/resources/application-connectiontest.yaml (hunk 3 above)
Comment (tanovellino) [reply dces-report-service/src/main/resources/application-connectiontest.yaml:57]: The comment is in the application.yaml. This is just for testing

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/config/DcesReportAuthenticationConfig.java (hunk 2 above)
Comment (tanovellino) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/config/DcesReportAuthenticationConfig.java:36]: will change.  I did not pay much attention being a 1 liner lambda function

**Review by LAA-Dave-Edwards — APPROVED** [review]

File: dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/config/DcesReportAuthenticationConfig.java (hunk 1 above)
Comment (tanovellino) [reply dces-report-service/src/main/java/uk/gov/justice/laa/crime/dces/report/config/DcesReportAuthenticationConfig.java:14]: I added the documentation of how to access the endpoint or triggering the report manually as part of the work needed to complete task DCES-77

**Pull request merged**